pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -303,55 +301,59 @@ func (exp *flowExporter) sendDataRecord(dataRec ipfix.IPFIXRecord, record flowex
+			ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, record.Conn.DestinationPodNamespace)
 		case "destinationPodName":
-			_, err = dataRec.AddInfoElement(ie, record.Conn.DestinationPodName)
+			ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, record.Conn.DestinationPodName)
 		case "destinationNodeName":
 			// Add nodeName for only local pods whose pod names are resolved.
 			if record.Conn.DestinationPodName != "" {
-				_, err = dataRec.AddInfoElement(ie, nodeName)
+				ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, nodeName)
 			} else {
-				_, err = dataRec.AddInfoElement(ie, "")
+				ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, "")
 			}
 		case "destinationClusterIPv4":
 			if record.Conn.DestinationServicePortName != "" {
-				_, err = dataRec.AddInfoElement(ie, record.Conn.TupleOrig.DestinationAddress)
+				ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, record.Conn.TupleOrig.DestinationAddress)
 			} else {
 				// Sending dummy IP as IPFIX collector expects constant length of data for IP field.
 				// We should probably think of better approach as this involves customization of IPFIX collector to ignore
 				// this dummy IP address.
-				_, err = dataRec.AddInfoElement(ie, net.IP{0, 0, 0, 0})
+				ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, net.IP{0, 0, 0, 0})
 			}
 		case "destinationServicePortName":
-			_, err = dataRec.AddInfoElement(ie, record.Conn.DestinationServicePortName)
+			if record.Conn.DestinationServicePortName != "" {
+				ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, record.Conn.DestinationServicePortName)
+			} else {
+				ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, "")
+			}
 		case "ingressNetworkPolicyName":
-			_, err = dataRec.AddInfoElement(ie, record.Conn.IngressNetworkPolicyName)
+			ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, record.Conn.IngressNetworkPolicyName)
 		case "ingressNetworkPolicyNamespace":
-			_, err = dataRec.AddInfoElement(ie, record.Conn.IngressNetworkPolicyNamespace)
+			ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, record.Conn.IngressNetworkPolicyNamespace)
 		case "egressNetworkPolicyName":
-			_, err = dataRec.AddInfoElement(ie, record.Conn.EgressNetworkPolicyName)
+			ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, record.Conn.EgressNetworkPolicyName)
 		case "egressNetworkPolicyNamespace":
-			_, err = dataRec.AddInfoElement(ie, record.Conn.EgressNetworkPolicyNamespace)
-		}
-		if err != nil {
-			return fmt.Errorf("error while adding info element: %s to data record: %v", ie.Name, err)
+			ieWithValue = ipfixentities.NewInfoElementWithValue(ie.Element, record.Conn.EgressNetworkPolicyNamespace)
 		}
+		elements = append(elements, ieWithValue)
 	}
 
-	sentBytes, err := exp.process.AddRecordAndSendMsg(ipfixentities.Data, dataRec.GetRecord())
+	dataSet.AddRecord(elements, templateID)
```

Comment:
don't we want to handle the error here? There can be elements whose values may have wrong format.

Reply:
Makes sense. Updated.